src/libopenrave/libopenrave.h

```diff
@@ -470,6 +470,43 @@ inline const char *strcasestr(const char *s, const char *find)
 }
 #endif
 
+
+/// \brief A local uid generator
+class UIDGenerator {
+public:
+    UIDGenerator() {
+        _prefix.clear();
+        _sUniqueId.clear();
+    }
+    UIDGenerator(std::string prefix): _prefix(prefix) {
+        _sUniqueId.clear();
+    }
+    virtual ~UIDGenerator() {}
+
+    /// \brief Check if the given id is unique, otherwise it will assign a unique one and udpate the set
+    void AssignIdIfNotUnique(std::string& id) {
+        if (!id.empty() && IsUnique(id)) {
+            _sUniqueId.insert(id);
```

Comment:
I am not sure about this implementation. You should:

1. add all the IDs already existing
2. assign new ID when they are empty

If I have a list `["", "body0", "body1"]` I don't want to have things shifted like `["body0", "body1", "body2"]`, I want `["body2", "body0", "body1"]`